gitbook/src/interacting-with-erigon/interacting-with-erigon.md

```diff
@@ -0,0 +1,163 @@
+```
+
+### WebSockets
+
+WebSockets is a bidirectional transport protocol. Most modern browsers support WebSockets.
+
+A WebSocket connection is maintained until it is explicitly terminated by either the client or the node.
+
+Because WebSockets are bidirectional, nodes can push events to clients, which enables clients to subscribe to specific events, such as new transactions in the transaction pool, and new logs for smart contracts.
+
+The configuration of the WebSocket server follows the same pattern as the HTTP server:
+
+* Enable it using `--ws`
+* Configure the server port by passing `--ws.port` (default `8546`) _node/nodecfg/defaults.go:34_
+* Configure cross-origin requests using `--ws.origins` (though this maps to `--http.corsdomain` in Erigon)
+* WebSocket APIs inherit from the HTTP API configuration
+
+```bash
+erigon --http --ws --http.api eth,net,debug,trace
+```
+
+### IPC
+
+IPC is a simpler transport protocol for use in local environments where the node and the client exist on the same machine.
+
+The IPC transport can be enabled using `--socket.enabled` and configured with `--socket.url`:
+
+```bash
+erigon --socket.enabled --socket.url unix:///var/run/erigon.ipc
+```
+
+On Linux and macOS, Erigon uses UNIX sockets. On Windows, IPC is provided using named pipes. The socket inherits the namespaces from `--http.api`.
+
+### gRPC
+
+Erigon also supports gRPC for high-performance access to blockchain data:
+
+```bash
+rpcdaemon --grpc --grpc.addr localhost --grpc.port 9090
+```
+
+#### GraphQL
+
+Erigon uses the standard GraphQL documented by Geth at [https://geth.ethereum.org/docs/interacting-with-geth/rpc/graphql](https://geth.ethereum.org/docs/interacting-with-geth/rpc/graphql).
+
+## Interacting with the RPC
+
+You can easily interact with these APIs just like you would with any Ethereum client.
+
+You can use `curl`, a programming language with a low-level library, or tools like Foundry to interact with the chain at the exposed HTTP or WebSocket port.
+
+To enable all APIs using an HTTP transport:
+
+```bash
+erigon --http --http.api "admin,debug,eth,erigon,net,trace,txpool,web3"
+```
+
+This allows you to then call:
+
+```bash
+curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}' localhost:8545
+
+# With cast (if using Foundry)
+cast block-number
+cast rpc admin_nodeInfo  
+cast rpc debug_traceTransaction <tx_hash>
+cast rpc erigon_forks
+```
+
+## Modular Architecture
+
+Erigon supports running components as separate processes. The RPC daemon can run independently from the main Erigon node:
+
+```bash
+# Start main Erigon node
+erigon --http=false --private.api.addr=localhost:9090
+
+# Start separate RPC daemon
+rpcdaemon --private.api.addr=localhost:9090 --http.api=eth,erigon,web3,net,debug,trace,txpool
+```
+
+This modular approach allows for better resource management and scalability.
```

Comment:
Maybe can add security note to the end: `admin_` and `debug_` namespaces are not intended to be public. Please make sure you don't expose RPC daemon port to public (to internet) - when using `admin_` and `debug_` namespaces. If need serve public RPC requests: we recommending start 2nd private RPCDaemon for `admin_`/`debug_` requests.